Services/GameStateService.cs

```diff
@@ -41,5 +53,80 @@ public PokerTable LoadGameStateFromFile(string filePath)
             string gameStateJson = File.ReadAllText(filePath);
             return DeserializeGameState(gameStateJson);
         }
+
+        //TODO learn about async and Task
+        public async Task<LoadGameResponse> LoadGameFromUrl(string url)
+        {
+            // Step 1: Fetch JSON from URL
+            var response = await _httpClient.GetAsync(url);
+            response.EnsureSuccessStatusCode();
+            var json = await response.Content.ReadAsStringAsync();
+            //Console.WriteLine(json);
+
+            // Step 2: Deserialize JSON to PokerTable Data
+            PokerTable pokerTable = JsonSerializer.Deserialize<PokerTable>(json, _jsonSerializerOptions);
+
+            // Step 3: Process PokerTable data
+            int entryStack = FindStartingStack(pokerTable);
+            var (playersAboveEntryStack, playersBehindEntryStack) = CategorizePlayers(pokerTable, entryStack);
+
+            return new LoadGameResponse
+            {
+                HighestStack = FindHighestStack(pokerTable),
+                LowestStack = FindLowestStack(pokerTable),
+                EntryStack = entryStack,
+                ListOfPlayersAboveEntryStack = playersAboveEntryStack,
+                ListOfPlayersBehindEntryStack = playersBehindEntryStack,
+                SuperFolksCount = FindSuperFolksCount(pokerTable)
+            };
+
+        }
+        private int FindStartingStack(PokerTable pokerTable)
+        {
+            int totalStack = pokerTable.Players.Sum(player => player.ChipCount);
+            int startingStack = totalStack / pokerTable.Players.Count();
+            return startingStack;
+        }
+
+        private PlayerStackInfo FindHighestStack(PokerTable pokerTable)
+        {
+            return pokerTable.Players
+                .Select(player => new PlayerStackInfo { Name = player.Name, Value = player.ChipCount })
```

Comment:
To jest ciekawe, bo to jest mocno nieoptymalne.

Zawsze się sprzeczam z Michaelem takim (co będzie rekrutować) jak to jest ważne (on uważa że mało :) ) Także jak tak zrobisz na interview to nikt się nie obrazi, ale.:


Co by było gdybyśmy mieli milion playerów :)

1. Select tworzy milion nowych obiektów, PlayerStackInfo
2. OrderByDescending sortuje milion obiektów (takie sortowanie jest mocno kosztowne)
3. Bierzesz jeden element, i olewasz 999'999.

Możnaby użyć MaxBy (wprowadzonego niedawno).

Do tego jakbyś najpierw zrobił ten "OrderBy" i "FirstOrDefault" a potem Select to funkcjonalnie byłoby to samo, a nie tworzylibyśmy nowych obiektów w takiej ilości, tylko wtedy jeden

Reply:
czy można to zrobić jeszcze lepiej niż to co poniżej (jakoś bez tworzenia zmiennej?)
var highestStackPlayer = pokerTable.Players.MaxBy(player => player.ChipCount);
return new PlayerStackInfo { Name = highestStackPlayer.Name, Value = highestStackPlayer.ChipCount };

Reply:
`return pokerTable.Players.MaxBy(player => player.ChipCount).Select( _ => new PlayerStackInfo {Name = _.Name, Value = _.ChipCount });`